iseq.c

```diff
@@ -1624,6 +1624,35 @@ iseqw_inspect(VALUE self)
     }
 }
 
+static ID iseq_type_ids[ISEQ_TYPE_PLAIN+1];
+
+static VALUE
+iseqw_type(VALUE self)
+{
+    const rb_iseq_t *iseq = iseqw_check(self);
+    const struct rb_iseq_constant_body *const body = ISEQ_BODY(iseq);
+
+    if (iseq_type_ids[0] == 0) {
+        iseq_type_ids[0] = rb_intern("top");
+        iseq_type_ids[1] = rb_intern("method");
+        iseq_type_ids[2] = rb_intern("block");
+        iseq_type_ids[3] = rb_intern("class");
+        iseq_type_ids[4] = rb_intern("rescue");
+        iseq_type_ids[5] = rb_intern("ensure");
+        iseq_type_ids[6] = rb_intern("eval");
+        iseq_type_ids[7] = rb_intern("main");
+        iseq_type_ids[8] = rb_intern("plain");
```

Comment:
Also `rb_intern` caches the interned `ID`, that is these `ID`s are doubly cached.
And the `enum` values would be better than magic numbers, I think.

Reply:
Ok, I'll fix it.  I guess we have the same problem [here](https://github.com/ruby/ruby/blob/6d069bdd418a450ab6d271844b6e26d6867eecfb/ext/objspace/objspace.c#L627-L651) too.